tests/foreman/api/test_errata.py

```diff
@@ -414,40 +420,72 @@ def package_applicability_changed_as_expected(
     return True
 
 
-def cv_publish_promote(sat, org, cv, lce=None, needs_publish=True):
+def cv_publish_promote(sat, org, cv, lce=None, needs_publish=True, force=False):
     """Publish & promote Content View Version with all content visible in org.
 
     :param lce: if None, default to 'Library',
-        pass a single instance of lce, or list of instances.
+        pass a single environment :id or instance.
+        Or pass a list of environment :ids or isntances.
     :param bool needs_publish: if False, skip publish of a new version
     :return dictionary:
         'content-view': instance of updated cv
         'content-view-version': instance of newest cv version
     """
     # Default to 'Library' lce, if None passed
-    # Take a single instance of lce, or list of instances
-    lce_ids = 'Library'
+    # Take a single lce id, of list of ids
+    # or Take a single instance of lce, or list of instances
+    lce_library = _fetch_library_environment_for_org(sat, org)
+    lce_ids = [lce_library.id]
     if lce is not None:
         if not isinstance(lce, list):
-            lce_ids = [lce.id]
+            # a list was not passed, just single lce or id
+            if isinstance(lce, int):
+                lce_ids = [lce]
+            else:
+                # will fail here if the single lce is not int or has no id
+                lce_ids = [lce.id]
         else:
-            lce_ids = [_lce.id for _lce in lce]
-
+            # a list of ids or instances was passed
+            if isinstance(lce[0], int):
+                lce_ids = lce
+            else:
+                # will fail here if list of lces is not int or has no id
+                lce_ids = [env.id for env in lce]
```

Comment:
```suggestion
            lce_ids = lce if isinstance(lce[0], int) else [env.id for env in lce]
```

Reply:
I implemented this, and the same logic for the non-list case above!